### PR TITLE
Dataset File Upload UI

### PR DESCRIPTION
**Pull request by tonyc856**
screen shot 2018-12-03 at 8 57 26 pm

JIRA ticket: https://jira.csc.calpoly.edu/browse/DAT-18

### CODE REVIEW

File: src/Upload/FilesList/FileItem.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+         <TableRow hover={true}>
+               <TableCell>
+                  {this.props.file.name}
+               </TableCell>
+               <TableCell>
+                  {this.props.file.size}   
+               </TableCell>
+               <TableCell>
+                  {this.props.file.type}        
+               </TableCell>
```
Comment (snayerman) [inline src/Upload/FilesList/FileItem.js]: Repetitive code to have `<TableCell>...</TableCell>` 3 times. Use Object.keys to map through all files for the cells

File: src/Upload/FilesList/FilesList.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Paper, Table, TableBody, TableHead, TableRow, TableCell } from '@material-ui/core';
+import FileItem from './FileItem'
+
+class FilesList extends React.Component {
```
Comment (snayerman) [inline src/Upload/FilesList/FilesList.js]: change to:
```
import React, { Component } from 'react';
...

export default class FilesList extends Component {
...
```

File: src/Upload/FilesList/FilesList.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+   renderFileItems() {
+      let files = Array.from(this.props.files);
```
Comment (snayerman) [inline src/Upload/FilesList/FilesList.js]: Is Array.from necessary? Isn't props.files already an array?

File: src/Upload/FilesList/FilesList.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      return (
+         <Paper className="files-list">
+               <Table>
+                  <TableHead>
+                     <TableRow>
+                           <TableCell>Filename</TableCell>
+                           <TableCell>Size</TableCell>
+                           <TableCell>Type</TableCell>
+                           <TableCell>Upload Progress</TableCell>
+                     </TableRow>
```
Comment (snayerman) [inline src/Upload/FilesList/FilesList.js]: When you have multiple components being rendered as siblings like the tablecells, you should move it out to a function like:

```
renderCells = (labels) => {
      return (
         <>
            {labels.map(label => {
               return <TableCell key={label}>{label}</TableCell>
            })}
         </>
      );
   }
```

and inside of `<TableRow>` have the following: `{this.renderCells(['Filename', 'Size', 'Type', 'Upload Progress'])}`

File: src/Upload/Upload.js [hunk 5]
```diff
@@ -24,12 +41,26 @@ export default class Upload extends Component {
...
          <div className="upload-body">
             UPLOAD DATASET
             <input type="file" onChange={this.handleChange}/>
 
             {this.state.selectedFile != null ? this.renderFileContents() : null}
          </div>
+>>>>>>> abb67e41c63a55910adb1c1825993abebe3c0a8e
       );
    }
 }
```
Comment (snayerman) [inline src/Upload/Upload.js]: This looks like an unresolved merge conflict?

File: src/Upload/FilesList/FilesList.js (hunk 3 above)
Comment (tonyc856) [reply src/Upload/FilesList/FilesList.js]: props.files gave back a FileList, which doesn't have map() like Arrays does so I converted it to an Array so I can use map to iterate thru each files.

File: src/Upload/FilesList/FileItem.js (hunk 1 above)
Comment (tonyc856) [reply src/Upload/FilesList/FileItem.js]: screen shot 2018-12-03 at 10 47 22 pm

Objects.keys(this.props.file) isn't returning any key strings for some reason. I ran it in console (look at above pic). This is so strange

File: src/Upload/Upload.js (hunk 5 above)
Comment (tonyc856) [reply src/Upload/Upload.js]: Well, that was embarrassing. I had it resolved in local but was didn't get pushed for some reason. I'll double check it gets pushed up.

File: src/Upload/FilesList/FilesList.js (hunk 2 above)
Comment (tonyc856) [reply src/Upload/FilesList/FilesList.js]: Can you briefly explain the benefit doing this over leaving a "export default FilesList" at the end of the file? It wasn't explain much in the udemy tutorial.

**Pull request merged**